docs/rfc/block-registration.md

```diff
@@ -0,0 +1,377 @@
+
+## Registering a block type
+
+To register a new block type, start by creating a `block.json` file. This file:
+
+* Gives a name to the block type.
+* Defines some important metadata about the registered block type (title, category, icon, description, keywords).
+* Defines the attributes of the block type.
+* Links to the editor implementation of your block type, the save function and any other context-aware property.
+
+**Example:**
+
+
+```json
+{
+	"name": "my-plugin/notice",
+	"title": "Notice",
+	"category": "common",
+	"icon": "star",
+	"description": "Shows warning, error or success notices  ...",
+	"keywords": [ "alert", "message" ],
+	"attributes": {
+		"message": {
+			"type": "string",
+			"source": "html",
+			"selector": ".meessage"
+		}
+	},
+	"edit": "blocks/notice-edit.js",
+	"save": "blocks/notice-save.js",
+}
+```
+
+## Block API
+
+This section describes all the properties that can be added to the `block.json` file to define the behavior and metadata of block types.
+
+### Name
+
+* Type: string
+* Required
+* Localized: No
+* Property: `name`
+
+```json
+{ "name": "core/heading" }
+```
+
+The name for a block is a unique string that identifies a block. Names have to be structured as `namespace/block-name`, where namespace is the name of your plugin or theme.
+
+**Note:** A block name can only contain lowercase alphanumeric characters, dashes, and at most one forward slash to designate the plugin-unique namespace prefix. It must begin with a letter.
+
+**Note:** This name is used on the comment delimiters as `<!-- wp:my-plugin/book -->`. Block types in the `core` namespace do not include a namespace when serialized.
+
+**Important Note:**
+
+Other block properties that point to JavaScript files use this identifier to provide values of the given properties.
+
+For instance, to define an SVG icon in a `block.js` file, you should attach it to the blocks global variable like so:
+
+```js
+blocks[ 'core/heading' ].icon = // SVG element of the icon. 
+```
+
+### Title
+
+* Type: string
+* Required
+* Localized: Yes
+* Property: `title`
+
+```json
+{ "title": "Heading" }
+```
+
+This is the display title for your block, which can be translated with our translation functions. The block inserter will show this name.
+
+### Category
+
+* Type: string
+* Required
+* Localized: No
+* Property: `category`
+
+```json
+{ "category": "common" }
+```
+
+Blocks are grouped into categories to help users browse and discover them.
+
+The core provided categories are:
+
+* common
+* formatting
+* layout
+* widgets
+* embed
+
+Plugins and Themes can also register [custom block categories](https://wordpress.org/gutenberg/handbook/designers-developers/developers/filters/block-filters/#managing-block-categories).
+
+An implementation should expect and tolerate unknown categories, providing some reasonable fallback behavior (e.g. a "common" category).
```

Comment:
So we have it covered. Nice. We will have to double check how strict block registration is at the moment. It's particularly important for the future work of WordPress.org blocks directory where you would be able to install plugins using the definition provided when using the inserter but you wouldn't have a given category registered as it is hidden inside PHP or JS code.